### PR TITLE
Jinja validation for incident tracking

### PR DESCRIPTION
**Pull request by diegocepedaw**
added jinja validation to incident tracking body

### CODE REVIEW

File: src/iris/api.py [hunk 1]
```diff
@@ -610,6 +610,13 @@ def is_valid_tracking_settings(t, k, tpl):
     else:
         if t not in cache.modes:
             return False, 'Unknown tracking type: %s' % t
+
+    environment = SandboxedEnvironment()
+    for app in tpl:
+        try:
+            environment.from_string(tpl[app]['body'])          
+        except Exception as e:
```
Comment (dwang159) [inline src/iris/api.py]: Catch `jinja2.TemplateSyntaxError`

File: src/iris/api.py [hunk 2]
```diff
@@ -610,6 +610,13 @@ def is_valid_tracking_settings(t, k, tpl):
     else:
         if t not in cache.modes:
             return False, 'Unknown tracking type: %s' % t
+
+    environment = SandboxedEnvironment()
```
Comment (dwang159) [inline src/iris/api.py]: This should go with the `else` in line 610, otherwise we validate email twice.

File: src/iris/api.py (hunk 1 above)
Comment (dwang159) [reply src/iris/api.py]: Probably update lines 601 and 608 too.

File: src/iris/api.py [hunk 3]
```diff
@@ -610,6 +610,13 @@ def is_valid_tracking_settings(t, k, tpl):
...
+    for app in tpl:
+        try:
+            environment.from_string(tpl[app]['body'])          
+        except Exception as e:
+                return False, 'Invalid jinja syntax in incident tracking text dy html: %s' % e
```
Comment (dwang159) [inline src/iris/api.py]: typo: "dy"

Comment (diegocepedaw) [reply src/iris/api.py]: fixed

File: src/iris/api.py (hunk 2 above)
Comment (diegocepedaw) [reply src/iris/api.py]: fixed

File: src/iris/api.py (hunk 1 above)
Comment (diegocepedaw) [reply src/iris/api.py]: updated as well as 601 608 and 595

**Pull request merged**